src/transformers/modeling_utils.py

```diff
@@ -497,7 +505,7 @@ def load_sharded_checkpoint(model, folder, strict=True, prefer_safe=True):
     return torch.nn.modules.module._IncompatibleKeys(missing_keys, unexpected_keys)
 
 
-def load_state_dict(checkpoint_file: Union[str, os.PathLike]):
+def load_state_dict(checkpoint_file: Union[str, os.PathLike], hf_quantizer=None):
```

Comment:
that is re-introducing quantizer related logic in the modeling utils, can't we pass `weights_only_kwarg` as arg here?

Reply:
I'll fix this

Reply:
Let me know if you prefer with these [changes](https://github.com/huggingface/transformers/pull/29023/commits/5fc659ce0c5008a842fe6821e30fd7b0e0f8eedd) ! Otherwise I can revert it.